pkgs/games/dynablaster-revenge/default.nix

```diff
@@ -0,0 +1,88 @@
+{ stdenv
+, lib
+, mkDerivation
+, fetchFromGitHub
+, qtbase
+, qmake
+, wrapQtAppsHook
+, libGLU
+, SDL2
+, alsa-lib
+
+, isServer ? false
+}:
+
+mkDerivation rec {
+  pname = "dynablaster-revenge"
+    + lib.optionalString isServer "-server";
+  version = "unstable-2021-01-02";
+
+  src = fetchFromGitHub {
+    owner = "varnholt";
+    repo = "dynablaster_revenge";
+    rev = "75538ac3f1080fbdf91c58b533e487ceb6eb2aec";
+    sha256 = "0qpdq8x3sjwgrqrfyfjwsf90rklf3lj0l0x7s8d0ss3yadq110la";
+  };
+
+  postUnpack = ''
+    export sourceRoot=$sourceRoot/${if isServer then "server" else "client"}
+  '';
+
+  postPatch = lib.optionalString isServer ''
+    substituteInPlace server.pro \
+      --replace 'TARGET = server' 'TARGET = ${meta.mainProgram}'
+  '';
+
+  nativeBuildInputs = [
+    qmake
+    wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    qtbase
+  ] ++ lib.optionals (!isServer) ([
+    libGLU
+    SDL2
+  ] ++ lib.optionals stdenv.isLinux [
+    alsa-lib
+  ]);
+
+  installPhase = ''
+    runHook preInstall
+
+  '' + (if stdenv.isDarwin then ''
+    mkdir -p $out/{Applications,bin}
+    mv {,$out/Applications/}${meta.mainProgram}.app
+    ln -s $out/{Applications/${meta.mainProgram}.app/Contents/MacOS,bin}/${meta.mainProgram}
+  '' else ''
+    install -Dm755 {,$out/bin/}${meta.mainProgram}
```

Comment:
Please hardcode or use pname.

Reply:
@OPNA2608 Maybe you are motivated to have a look at this?